ets-demo/etstool.py

```diff
@@ -0,0 +1,361 @@
+<http://docs.enthought.com/edm/>`_
+
+To use this to run you tests, you will need to install EDM and click
+into your working environment.  You will also need to have git
+installed to access required source code from github repositories.
+You can then do::
+
+    python etstool.py install --runtime=... --toolkit=...
+
+to create a test environment from the current codebase and::
+
+    python etstool.py test --runtime=... --toolkit=...
+
+to run tests in that environment.  You can remove the environment with::
+
+    python etstool.py cleanup --runtime=... --toolkit=...
+
+If you make changes you will either need to remove and re-install the
+environment or manually update the environment using ``edm``, as
+the install performs a ``python setup.py install`` rather than a ``develop``,
+so changes in your code will not be automatically mirrored in the test
+environment.  You can update with a command like::
+
+    edm run --environment ... -- python setup.py install
+
+You can run all three tasks at once with::
+
+    python etstool.py test_clean --runtime=... --toolkit=...
+
+which will create, install, run tests, and then clean-up the environment.  And
+you can run tests in all supported runtimes and toolkits (with cleanup)
+using::
+
+    python etstool.py test_all
+
+Currently supported runtime values are ``2.7`` and ``3.5``, and currently
+supported toolkits are ``null``, ``pyqt``, ``pyside`` and ``wx``.  Not all
+combinations of toolkits and runtimes will work, but the tasks will fail with
+a clear error if that is the case.
+
+Tests can still be run via the usual means in other environments if that suits
+a developer's purpose.
+
+Changing This File
+------------------
+
+To change the packages installed during a test run, change the dependencies
+variable below.  To install a package from github, or one which is not yet
+available via EDM, add it to the `ci-src-requirements.txt` file (these will be
+installed by `pip`).
+
+Other changes to commands should be a straightforward change to the listed
+commands for each task. See the EDM documentation for more information about
+how to run commands within an EDM enviornment.
+
+"""
+
+import glob
+import os
+import subprocess
+import sys
+from shutil import rmtree, copy as copyfile
+from tempfile import mkdtemp
+from contextlib import contextmanager
+
+import click
+
+supported_combinations = {
+    '3.5': {'pyside2', 'pyqt', 'pyqt5', 'null'},
+    '3.6': {'pyside2', 'pyqt', 'pyqt5', 'wx', 'null'},
+}
+
+# Default Python version to use in the comamnds below if none is specified.
+DEFAULT_RUNTIME = '3.6'
+
+# Default toolkit to use if none specified.
+DEFAULT_TOOLKIT = 'null'
+
+# The main package name, also used to form the Python environment name.
+PACKAGE_NAME = "etsdemo"
+
+dependencies = {
+    "configobj",
+    "docutils",
+    "flake8",
+    "traits",
+    "traitsui",
+    "pyface",
+    "pip",
+}
+
+extra_dependencies = {
+    'pyside': {'pyside'},
```

Comment:
Well spotted!  I was trying to avoid interfering with #821 and has clearly failed :D

Reply:
Sorry if I wasn't clear - `pyside` should be removed from extra dependencies because it's not supported. Support for it shouldn't be added.

Reply:
But when we proceed with bringing PyQt4 and PySide back to CI for TraitsUI, this distribution will be affected because this distribution is still tied with TraitsUI on CI. There is a potential for PR interactions when we modify CI build configurations. So I think the safest thing to do is to let this distribution support all the backends we are going to support once #815 is fixed.

Reply:
Sorry, been referencing the PR, I should be referring to #821

Now I looked at again, `"pyside"` was not in `supported_combinations` when #815 removed support for PySide, is that what you meant?

Reply:
Because it does not support Python 3, of course. Yep, it should be removed.